lib/survey_gizmo/resource.rb

```diff
@@ -39,27 +39,31 @@ def all(conditions = {}, _deprecated_filters = {})
         conditions[:resultsperpage] = SurveyGizmo.configuration.results_per_page unless conditions[:resultsperpage]
 
         request_route = handle_route!(:create, conditions)
-        response = RestResponse.new(SurveyGizmo.get(request_route + filters_to_query_string(conditions)))
-        collection = response.data.map { |datum| datum.is_a?(Hash) ? new(datum) : datum }
-
-        while all_pages && response.current_page < response.total_pages
-          paged_filter = filters_to_query_string(conditions.merge(page: response.current_page + 1))
-          response = RestResponse.new(SurveyGizmo.get(request_route + paged_filter))
-          collection += response.data.map { |datum| datum.is_a?(Hash) ? new(datum) : datum }
-        end
-
-        # Add in the properties from the request because many of the important ones (like survey_id) are
-        # not often part of the SurveyGizmo returned data
-        properties.each do |k, v|
-          next unless v && instance_methods.include?(k)
-          collection.each { |c| c[k] ||= v }
-        end
-
-        # Sub questions are not pulled by default so we have to retrieve them manually
-        # SurveyGizmo claims they will fix this bug and eventually all questions will be
-        # returned in one request.
-        if self == SurveyGizmo::API::Question
-          collection += collection.map { |question| question.sub_questions }.flatten
+        collection = []
+        response = nil
+
+        while !response || (all_pages && response.current_page < response.total_pages)
+          paged_filter = filters_to_query_string(conditions.merge(page: response ? response.current_page + 1 : 1))
+          Pester.survey_gizmo_ruby.retry do
```

Comment:
If we were using faraday instead of HTTParty, it would be trivial to wrap all requests.  Oh well.